### PR TITLE
Update metadata endpoints response types

### PR DESCRIPTION
**Pull request by katia-qm**
**Background**
Metadata creation endpoints have changed their response types from None.

**Changes**
- created response types for metadata endpoints
- updates parser to work with parametrized generics

### CODE REVIEW

File: hari_client/client/client.py [hunk 1]
```diff
@@ -1273,5 +1318,7 @@ def create_crops(
             f"/datasets/{dataset_id}/crops",
             params={"subset_id": subset_id},
             json=self._pack(locals(), ignore=["dataset_id", "subset_id"]),
-            success_response_item_model=None,
+            success_response_item_model=list[
+                models.UpdateHistogramsResponse | models.CreateCropsResponse
```
Comment (Dens49) [inline hari_client/client/client.py:1322]: our minimum python version is 3.9, but using the pipe symbol in the typehint isn't supported in python 3.9. Can you switch to using typing.Union?

**Review by Dens49 — CHANGES_REQUESTED** [review]
Only 2 change requests:
- minimum python version we support is 3.9, but using the pipe for a union typehint isn't supported in python 3.9. All the new models have to be updated
- can you explain the support of the union in the docstring of `_parse_response_model` along with its limitations? For example the limits on nesting

File: hari_client/client/client.py [hunk 2]
```diff
@@ -1312,13 +1325,18 @@ def create_crops(
 
         Raises:
             APIException: If the request fails.
+
+        Returns:
+            list[models.UpdateHistogramsResponse, models.CreateCropsResponse]: list of updated histograms and created crops
```
Comment (Dens49) [inline hari_client/client/client.py:1330]:
```suggestion
            list[typing.Union[models.UpdateHistogramsResponse, models.CreateCropsResponse]]: list of updated histograms and created crops
```

Comment (katia-qm) [reply hari_client/client/client.py:1330]: oops, thanks for finding this!

**Review by Dens49 — APPROVED** [review]
looks good! just a tiny update for one of the docstrings

**Pull request merged**